stdlib/public/core/IntegerTypes.swift.gyb

```diff
@@ -1160,7 +1160,7 @@ public struct ${Self}
   public init(_ source: ${FloatType}) {
     _precondition(source.isFinite,
       "${FloatType} value cannot be converted to ${Self} because it is either infinite or NaN")
-%     if not (FloatBits == 16 and bits >= 32): # Float16 is always in-range for 32- and 64-bit ints.
+%     if not (FloatBits == 16 and signed and bits >= 32): # Float16 is always in-range for 32- and 64-bit signed integer types.
```

Comment:
We could tighten this up a bit more and elide the upper-bound check for u16, u32, u64.

Reply:
Fiiiiiine...